loopy/tools.py

```diff
@@ -601,4 +610,124 @@ def is_interned(s):
+        [
+        % for dom in kernel.domains:
+        "${str(dom)}",
+        % endfor
+        ],
+        '''
+        % for id, opts in options.items():
+        <% insn = kernel.id_to_insn[id] %>
+        % if isinstance(insn, lp.Assignment):
+        ${insn.assignee} = ${insn.expression} {${opts}}
+        % elif isinstance(insn, lp.BarrierInstruction):
+        ... ${insn.synchronization_kind[0]}barrier{${opts}}
+        % elif isinstance(insn, lp.NoOpInstruction):
+        ... nop {${opts}}
+        % else:
+        <% raise NotImplementedError("Not implemented for ${type(insn)}.")%>
+        % endif
+        %endfor
+        ''', [
+            % for arg in kernel.args:
+            % if isinstance(arg, lp.ValueArg):
+            lp.ValueArg(
+                name="${arg.name}",
+                dtype=${('np.'+arg.dtype.numpy_dtype.name
+                            if arg.dtype else 'lp.auto')}),
+            % else:
+            lp.GlobalArg(
+                name="${arg.name}", dtype=${('np.'+arg.dtype.numpy_dtype.name
+                                                if arg.dtype else 'lp.auto')},
+                shape=${arg.shape}, for_atomic=${arg.for_atomic}),
+            % endif
+            % endfor
+            % for tv in kernel.temporary_variables.items():
+            lp.TemporaryVariable(
+                name="${tv.name}",
+                dtype=${'np.'+tv.dtype.numpy_dtype.name if arg.dtype else 'lp.auto'},
+                shape=${tv.shape}, for_atomic=${tv.for_atomic},
+                address_space=${tv_scope[tv.address_space]},
+                read_only=${tv.read_only},
+                % if tv.initializer is not None:
+                initializer=${"np."+repr(tv.initializer)},
+                % endif
+                ),
+            % endfor
+            ],
+            lang_version=${lp.MOST_RECENT_LANGUAGE_VERSION},
+    % if kernel.iname_slab_increments:
+            iname_slab_increments=${repr(kernel.iname_slab_increments)},
+    % endif
+    % if kernel.applied_iname_rewrites:
+            applied_iname_rewrites=${repr(kernel.applied_iname_rewrites)},
+    % endif
+    % if kernel.name != "loopy_kernel":
+            name="${kernel.name}",
+    % endif
+            )
+
+    % for iname, tags in kernel.iname_to_tags.items():
+    % for tag in tags:
+    kernel = lp.tag_inames(kernel, "${"%s:%s" %(iname, tag)}")
+    % endfor
+    % endfor
+    """
+
+    python_code = Template(python_code).render(options=options,
+            kernel=kernel)
+    python_code = remove_lines_with_only_spaces(
+            remove_common_indentation(python_code))
+
+    # {{{ checking whether the written string is accurate
+
+    reproducer_variables = {}
+    exec(python_code, reproducer_variables)
+    if reproducer_variables["kernel"] != kernel:
+        # knl1 = kernel
+        # knl2 = reproducer_variables["kernel"]
+        raise LoopyError(f"Error generating reproducer for '{kernel.name}'.")
+
+    # }}}
```

Comment:
Package into a function and use assert?

Reply:
Moved to a function, kept it as a LoopyError 69fc8a8.